src/test/bin/winkernel/control.cpp

```diff
@@ -222,6 +342,26 @@ QuicTestCtlUninitialize(
 
     delete MsQuic;
 
+    if (InterlockedFetchAndSetBoolean(&NmrClient.Deleting)) {
+        //
+        // We are already in the middle of detaching the client.
+        // Complete it now.
+        //
+        NmrClientDetachProviderComplete(NmrClient.NmrClientHandle);
+    }
+
+    if (NmrClient.NmrClientHandle) {
+        NTSTATUS Status = NmrDeregisterClient(NmrClient.NmrClientHandle);
+        CXPLAT_FRE_ASSERTMSG(Status == STATUS_PENDING, "client deregistration failed");
```

Comment:
Should we really bugcheck if this fails? Can OOM trigger this failure? What then?

Reply:
NmrDeregisterClient strictly only returns STATUS_PENDING. This is sort of a dummy check.